include/fiction/utils/layout_utils.hpp

```diff
@@ -224,6 +227,123 @@ template <typename Lyt>
+        {
+            if (c.y <= y_min && c.x <= x_min)
+            {
+                y_min = c.y;
+                x_min = c.x;
+            }
+            else if (c.y <= y_min)
+            {
+                y_min = c.y;
+            }
+            else if (c.x <= x_min)
+            {
+                x_min = c.x;
+            }
+        });
+
+    lyt.foreach_cell(
+        [&lyt_new, &lyt, &x_min, &y_min](const auto& c)
+        {
+            lyt_new.assign_cell_type({c.x - x_min, c.y - y_min}, lyt.get_cell_type(c)),
+                lyt_new.assign_cell_mode({c.x - x_min, c.y - y_min}, lyt.get_cell_mode(c)),
+                lyt_new.assign_cell_name({c.x - x_min, c.y - y_min}, lyt.get_cell_name(c));
+        });
+    lyt_new.set_layout_name(lyt.get_layout_name());
+    return lyt_new;
+}
+
+/**
+ * The cell coordinates of a given layout are converted to SiQAD coordinates. A new layout with SiQAD coordinates is
+ * returned.
+ *
+ * @tparam Lyt Cell-level layout based on Fiction coordinates (Cube, Offset).
+ * @param lyt The given layout which is converted to a new layout based on SiQAD coordinates.
+ * @return new layout based on SiQAD coordinates.
+ */
+template <typename Lyt>
+sidb_cell_clk_lyt_siqad lyt_coordinates_to_siqad(const Lyt& lyt)
+{
+    static_assert(is_cartesian_layout_v<Lyt>, "Lyt is not a cartesian layout");
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout");
+
+    sidb_cell_clk_lyt_siqad lyt_new{};
+    lyt.foreach_cell(
+        [&lyt_new, &lyt](const auto& c)
+        {
+            lyt_new.assign_cell_type(siqad::to_siqad_coord<cell<Lyt>>(c), lyt.get_cell_type(c)),
+                lyt_new.assign_cell_mode(siqad::to_siqad_coord<cell<Lyt>>(c), lyt.get_cell_mode(c)),
+                lyt_new.assign_cell_name(siqad::to_siqad_coord<cell<Lyt>>(c), lyt.get_cell_name(c));
+        });
+    lyt_new.set_layout_name(lyt.get_layout_name());
+
+    return lyt_new;
+}
+
+/**
+ * The cell coordinates of a given layout are converted to Fiction coordinates. A new layout with Fiction coordinates is
+ * returned.
+ *
+ * @tparam Lyt Cell-level layout based on Fiction coordinates.
+ * @param lyt The given layout which is converted to a new layout based on Fiction coordinates.
+ * @return New layout based on Fiction coordinates.
+ */
+template <typename Lyt>
+Lyt lyt_coordinates_to_fiction(const sidb_cell_clk_lyt_siqad& lyt)
+{
+    static_assert(is_cartesian_layout_v<Lyt>, "Lyt is not a cartesian layout");
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout");
+
+    Lyt lyt_new{};
+
+    if constexpr (has_offset_ucoord_v<Lyt>)
+    {
+        auto lyt_normalized = normalize_layout_coordinates<sidb_cell_clk_lyt_siqad>(lyt);
+        lyt_normalized.foreach_cell(
+            [&lyt_new, &lyt_normalized](const auto& c)
```

Comment:
To minimize code duplication, you can make this lambda passed to `foreach_cell` into a variable and pass it in both branches

Reply:
is this possible even though the capture is different?

Reply:
Couldn't you just assign `lyt_new = normalize_layout_coordinates(...);` in case of `has_offset_ucoord_v`? And then use the same call?